src/sentry/utils/assets.py

```diff
@@ -1,22 +1,55 @@
 from __future__ import annotations
 
 import os.path
+from dataclasses import dataclass
 
 from cachetools.func import ttl_cache
 from django.conf import settings
 
 from sentry.utils import json
 
 
+@dataclass
+class FrontendVersions:
+    commit_sha: str | None
+    """
+    The commit SHA of the currently deployed frontend version.
+
+    XXX(epuirkhiser): currently may be None while we transition to a
+    frontend-versions file that contains the version_sha.
+    """
+    entrypoints: dict[str, str]
+    """
+    A mapping of unversioned entrypoint names to versioned entrypoints,
+    containing a content-hash suffix.
+    """
+
+
 @ttl_cache(ttl=60)
-def _frontend_versions() -> dict[str, str]:
+def _frontend_versions() -> FrontendVersions | None:
+    config = os.path.join(settings.CONF_DIR, "settings", "frontend", "frontend-versions.json")
     try:
-        with open(
-            os.path.join(settings.CONF_DIR, "settings", "frontend", "frontend-versions.json")
-        ) as f:
-            return json.load(f)  # getsentry path
+        with open(config) as f:
+            contents = json.load(f)  # getsentry path
     except OSError:
-        return {}  # common case for self-hosted
+        return None  # common case for self-hosted
+
+    # XXX(epurkhiser): Shim code while we transition to a
+    # `frontend-versions.json` file that contains both a webpack entrypoint
+    # mapping as well as a commit SHA.
+    if "commit_sha" not in contents:
+        return FrontendVersions(commit_sha=None, entrypoints=contents)
+
+    return FrontendVersions(**contents)
+
+
+def get_frontend_commit_sha() -> str | None:
+    """
+    Returns the commit SHA of the currently configured frontend-versions.
+    """
+    if versions := _frontend_versions():
```

Comment:
what's this `:=` operator?

Reply:
Assignment expression operator (aka walrus operator)

https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions